app/services/create_collection.rb

```diff
@@ -0,0 +1,40 @@
+class CreateCollection
+  Result = Struct.new(:successful?, :collection, :errors)
```

Comment:
```suggestion
  include Service
```

We have a Service module that handles all this